Update changelog (#486)
hashicorp/consul-k8s: CHANGELOG.md

## UNRELEASED

> BREAKING CHANGES:

Author: part of this diff is putting breaking changes at the top

  Example Service:
  ```yaml
  ---
  apiVersion: v1
  kind: Service
  metadata:
    name: sample-app
  spec:
    selector:
      app: sample-app
    ports:
      - port: 80
        targetPort: 9090
  ---
  apiVersion: apps/v1
  kind: Deployment
  metadata:
    labels:
      app: sample-app
    name: sample-app
  spec:
    replicas: 1
    selector:
       matchLabels:
         app: sample-app
    template:
      metadata:
        annotations:
          'consul.hashicorp.com/connect-inject': 'true'
        labels:
          app: sample-app
      spec:
        containers:
        - name: sample-app
          image: sample-app:0.1.0
          ports:
          - containerPort: 9090
    ```
* Connect: `consul.hashicorp.com/connect-sync-period` annotation is no longer supported.
  This annotation was used to configure the sync period of the `consul-sidecar` (aka `lifecycle-sidecar`).
  Since we no longer inject the `consul-sidecar` to keep services registered in Consul, this annotation is
  now meaningless. [[GH-467](https://github.com/hashicorp/consul-k8s/pull/467)]
* Connect: transparent proxy feature is enabled by default. This may break existing deployments.

>   Please details of the feature below.

Reviewer: ```suggestion
  Please see details of the feature below.
```

>   Please see details of the feature below.

Reviewer 2: nice addition!

FEATURES:
* Connect: Support Transparent Proxy. [[GH-481](https://github.com/hashicorp/consul-k8s/pull/481)]
  This feature enables users to use KubeDNS to reach other services within the Consul Service Mesh,
  as well as enforces the inbound and outbound traffic to go through the Envoy proxy.
  Using transparent proxy for your service mesh applications means:
  - Proxy service registrations will set `mode` to `transparent` in the proxy configuration
    so that Consul can configure the Envoy proxy to have an inbound and outbound listener.
  - Both proxy and service registrations will include the cluster IP and service port of the Kubernetes service
    as tagged addresses so that Consul can configure Envoy to route traffic based on that IP and port.
  - The `consul-connect-inject-init` container will run `consul connect redirect-traffic` [command](https://www.consul.io/commands/connect/redirect-traffic),
    which will apply rules (via iptables) to redirect inbound and outbound traffic to the proxy.

>     To run this command the `consul-connect-inject-init` requires to run as root and have `NET_ADMIN`

Reviewer: ```suggestion
    To run this command the `consul-connect-inject-init` requires running as root with capability `NET_ADMIN`
```
Not sure how to word this, or if my suggestion is any better :D

Author: love it!
